### ISSUE TITLE
[Bug] omegaconf is doing interpolation during merges when it shouldnt 

### ISSUE THREAD

**Issue by alexeib** [omry/omegaconf#431]
copied from hydra issue here:
https://github.com/facebookresearch/hydra/issues/1125

repro:

```python
from omegaconf import OmegaConf
from dataclasses import dataclass
from typing import List
@dataclass
class test_class:
    x: List[float] = "${optimization.lr}"

a = OmegaConf.structured(test_class())
b = OmegaConf.create({"x": [0.1]})
OmegaConf.merge(a,b)
```
Exception:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/private/home/abaevski/.conda/envs/fairseq-fp16-20200821/lib/python3.6/site-packages/omegaconf/omegaconf.py", line 321, in merge
    target.merge_with(*others[1:])
  File "/private/home/abaevski/.conda/envs/fairseq-fp16-20200821/lib/python3.6/site-packages/omegaconf/basecontainer.py", line 327, in merge_with
    self._format_and_raise(key=None, value=None, cause=e)
  File "/private/home/abaevski/.conda/envs/fairseq-fp16-20200821/lib/python3.6/site-packages/omegaconf/base.py", line 101, in _format_and_raise
    type_override=type_override,
  File "/private/home/abaevski/.conda/envs/fairseq-fp16-20200821/lib/python3.6/site-packages/omegaconf/_utils.py", line 675, in format_and_raise
    _raise(ex, cause)
  File "/private/home/abaevski/.conda/envs/fairseq-fp16-20200821/lib/python3.6/site-packages/omegaconf/_utils.py", line 591, in _raise
    raise ex  # set end OC_CAUSE=1 for full backtrace
omegaconf.errors.ConfigKeyError: str interpolation key 'optimization.lr' not found
	full_key:
	reference_type=Any
	object_type=test_class
```

if instead you set a to be DictConfig initialized from a dict, everything works
```
>>> a = DictConfig({'x': '${optimization.lr}'})
>>> OmegaConf.merge(a,b)
{'x': [0.1]}
```

**Comment by omry** [comment]
@pereman2, can you take a look?
This should not resolve the interpolation.

I would like this fixed for 2.0 as well.

**Comment by pereman2** [comment]
Sure
